File: algorithms/swift/35-search-insert-position.swift
// https://leetcode.com/problems/search-insert-position/

class Solution {
    func searchInsert(_ nums: [Int], _ target: Int) -> Int {
        var l = 0, r = nums.count-1
        while(l <= r) {
            var mid: Int = (l+r)/2
            if (nums[mid]==target) { return mid }
            if (nums[mid] > target) { r = mid-1 }
            else { l = mid+1 }
        }
        return l
    }
}

File: algorithms/c++/35-search-insert-position.cpp
// https://leetcode.com/problems/search-insert-position/
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int left = 0;
        int right = nums.size() - 1;
        
        while(left <= right) {
            int mid = (left + right)/2;
            if (nums[mid] == target)
                return mid;
            if (nums[mid] > target)
                right = mid - 1;
            else
                left = mid + 1;
        }
        return left;
    }
};

int main() {

    return 0;
}